boltstub_rs/src/values/bolt_struct/unknown_type.rs

```diff
@@ -0,0 +1,140 @@
+use std::fmt::{Debug, Display, Formatter};
+
+use serde_json::Value as JsonValue;
+
+use super::_parsing::{check_last_json_field, next_json_field, next_pack_stream_field};
+use crate::bolt_version::JoltVersion;
+use crate::parse_error::ParseError;
+use crate::parser::ActorConfig;
+use crate::values::bolt_struct::TAG_UNKNOWN_TYPE;
+use crate::values::pack_stream_value::{PackStreamStruct, PackStreamValue};
+use indexmap::IndexMap;
+
+#[derive(Debug, Clone)]
+pub(crate) struct JoltUnknownType {
+    pub(crate) name: String,
+    pub(crate) minimum_protocol_major: i64,
+    pub(crate) minimum_protocol_minor: i64,
+    pub(crate) message: Option<String>,
+}
+
+impl JoltUnknownType {
+    pub(crate) fn parse(
+        v: JsonValue,
+        jolt_version: JoltVersion,
+        config: &ActorConfig,
+    ) -> Result<Self, ParseError> {
+        if !matches!(jolt_version, JoltVersion::V3) {
+            return Err(ParseError::new(format!(
+                "Sigil \"UT\" (unknown type) can only be parsed in JoltVersion::V3, using {jolt_version:?}"
+            )));
+        }
```

Comment:
Either (admittedly ugly)
```suggestion
        match jolt_version {
            JoltVersion::V1 | JoltVersion::V2 => {
                return Err(ParseError::new(format!(
                    "Sigil \"UT\" (unknown type) can only be parsed in JoltVersion::V3, \
                    using {jolt_version:?}"
                )));
            }
            JoltVersion::V3 => {}
        }
```
or 
```suggestion
        if jolt_version < JoltVersion::V3 {
            return Err(ParseError::new(format!(
                "Sigil \"UT\" (unknown type) can only be parsed in JoltVersion::V3, \
                    using {jolt_version:?}"
            )));
        }
```
To make the code more future proof. The former will cause a compiler error when a new Jolt version is introduced, the latter assumes that future versions are unlikely to change the semantics of the unknown type. The latter also requires to derive `PartialOrd` (and while you're at it Ord` wouldn't hurt either) for `JoltVersion`.